File: myai-java-service/src/main/java/com/test/login/dto/KnowledgeQueryRequest.java
package com.test.login.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class KnowledgeQueryRequest {
    @JsonProperty("user_id")
    private String userId;

    private String query;

    @JsonProperty("top_k")
    private Integer topK = 3;

    @JsonProperty("file_id")
    private String fileId;

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getQuery() {
        return query;
    }

    public void setQuery(String query) {
        this.query = query;
    }

    public Integer getTopK() {
        return topK;
    }

    public void setTopK(Integer topK) {
        this.topK = topK;
    }

    public String getFileId() {
        return fileId;
    }

    public void setFileId(String fileId) {
        this.fileId = fileId;
    }
}

File: myai-python-agent/app/schemas/knowledge_query_request.py
from typing import Literal

from pydantic import BaseModel


class KnowledgeQueryRequest(BaseModel):
    user_id: str
    query: str
    top_k: int = 3
    file_id: str | None = None
    retrieval_strategy: Literal["vector", "hybrid", "hybrid_rerank"] = "hybrid_rerank"
